source/extensions/omni.isaac.lab/omni/isaac/lab/controllers/operational_space.py

```diff
@@ -185,43 +134,83 @@ def reset_idx(self, robot_ids: torch.Tensor = None):
-            task_space_command, stiffness = torch.tensor_split(command, (self.target_dim, 6), dim=-1)
+            task_space_command, stiffness = torch.split(command, (self.target_dim, 6), dim=-1)
             # format command
-            stiffness = stiffness.clip_(min=self._p_gains_limits[0], max=self._p_gains_limits[1])
+            stiffness = stiffness.clip_(min=self._p_gains_limits[..., 0], max=self._p_gains_limits[..., 1])
             # joint positions + stiffness
             self._task_space_target[:] = task_space_command.squeeze(dim=-1)
             self._p_gains[:] = stiffness
-            self._d_gains[:] = 2 * torch.sqrt(self._p_gains)  # critically damped
+            self._d_gains[:] = 2 * torch.sqrt(self._p_gains) * torch.tensor(self.cfg.damping_ratio, device=self._device)
         elif self.cfg.impedance_mode == "variable":
             # split input command
-            task_space_command, stiffness, damping_ratio = torch.tensor_split(command, 3, dim=-1)
+            task_space_command, stiffness, damping_ratio = torch.split(command, (self.target_dim, 6, 6), dim=-1)
             # format command
-            stiffness = stiffness.clip_(min=self._p_gains_limits[0], max=self._p_gains_limits[1])
-            damping_ratio = damping_ratio.clip_(min=self._damping_ratio_limits[0], max=self._damping_ratio_limits[1])
+            stiffness = stiffness.clip_(min=self._p_gains_limits[..., 0], max=self._p_gains_limits[..., 1])
+            damping_ratio = damping_ratio.clip_(
+                min=self._damping_ratio_limits[..., 0], max=self._damping_ratio_limits[..., 1]
+            )
             # joint positions + stiffness + damping
             self._task_space_target[:] = task_space_command
             self._p_gains[:] = stiffness
             self._d_gains[:] = 2 * torch.sqrt(self._p_gains) * damping_ratio
         else:
             raise ValueError(f"Invalid impedance mode: {self.cfg.impedance_mode}.")
 
+        # resolve the target commands
+        target_groups = torch.split(self._task_space_target, self.target_list, dim=1)
+        for command_type, target in zip(self.cfg.command_types, target_groups):
+            if command_type == "position_rel":
+                # check input is provided
+                if ee_pose is None:
+                    raise ValueError("End-effector pose is required for 'position_rel' command.")
+                # scale command
+                target @= self._position_command_scale
+                # compute targets
+                self.desired_ee_pos = ee_pose[:, :3] + target
+                self.desired_ee_rot = ee_pose[:, 3:]
+            elif command_type == "position_abs":
+                # check input is provided
+                if ee_pose is None:
+                    raise ValueError("End-effector pose is required for 'position_abs' command.")
+                # compute targets
+                self.desired_ee_pos = target
+                self.desired_ee_rot = ee_pose[:, 3:]
```

Comment:
Similar question as in position_rel

Reply:
These are now removed (please see the comment above).